models/chainlink/arbitrum/chainlink_arbitrum_fm_fulfilled_transactions.sql

```diff
@@ -0,0 +1,65 @@
+{{
+  config(
+    tags=['dunesql'],
+    alias=alias('fm_fulfilled_transactions'),
+    partition_by=['date_month'],
+    materialized='incremental',
+    file_format='delta',
+    incremental_strategy='merge',
+    unique_key=['tx_hash', 'tx_index', 'node_address'],
+    post_hook='{{ expose_spells(\'["arbitrum"]\',
+                                "project",
+                                "chainlink",
+                                \'["linkpool_jon"]\') }}'
+  )
+}}
+
+{% set incremental_interval = '7' %}
+
+WITH
+  arbitrum_usd AS (
+    SELECT
+      minute as block_time,
+      price as usd_amount
+    FROM
+      {{ source('prices', 'usd') }} price
+    WHERE
+      symbol = 'ETH'
+      {% if is_incremental() %}
+        AND minute >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
+      {% endif %}      
+  ),
+  fm_submission_transactions AS (
+    SELECT
+      tx.hash as tx_hash,
+      tx.index as tx_index,
+      MAX(tx.block_time) as block_time,
+      cast(date_trunc('month', MAX(tx.block_time)) as date) as date_month,
+      tx."from" as "node_address",
+      MAX(
+        (cast((gas_used) as double) / 1e18) * effective_gas_price
+      ) as token_amount,
+      MAX(arbitrum_usd.usd_amount) as usd_amount
+    FROM
+      {{ source('arbitrum', 'transactions') }} tx
+      RIGHT JOIN {{ ref('chainlink_arbitrum_fm_gas_submission_logs') }} fm_gas_submission_logs ON fm_gas_submission_logs.tx_hash = tx.hash
```

Comment:
it would be beneficial to add a join condition here for the normal incremental filter on last 7 days, as this is a view on top of a large base table

Reply:
assume this applies universally across all blockchain versions

Reply:
As in add a conditional to the left join like this?
`{% if is_incremental() %}
        AND arbitrum_usd.block_time >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
      {% endif %}`

Reply:
yep, exactly, but i mean on the table being right joined:
`{% if is_incremental() %} AND fm_gas_submission_logs.block_time >= date_trunc('day', now() - interval '{{incremental_interval}}' day) {% endif %}`

Reply:
ah sorry I gotcha. Getting it updated now.